CSharp/AccountManagement/Program.cs

```diff
@@ -1,334 +1,349 @@
+
+                // Get account keys
+                Console.WriteLine("Getting account keys of account {0}...", account.Name);
+                BatchAccountListKeyResult accountKeys = await batchManagementClient.Account.ListKeysAsync(ResourceGroupName, account.Name);
+                Console.WriteLine("  Primary key of account {0}:   {1}", account.Name, accountKeys.Primary);
+                Console.WriteLine("  Secondary key of account {0}: {1}", account.Name, accountKeys.Secondary);
+                Console.WriteLine();
+
+                // Regenerate primary account key
+                Console.WriteLine("Regenerating the primary key of account {0}...", account.Name);
+                BatchAccountRegenerateKeyResult newKeys = await batchManagementClient.Account.RegenerateKeyAsync(
+                    ResourceGroupName, account.Name,
+                    new BatchAccountRegenerateKeyParameters() { KeyName = AccountKeyType.Primary });
+                Console.WriteLine("  New primary key of account {0}: {1}", account.Name, newKeys.Primary);
+                Console.WriteLine("  Secondary key of account {0}:   {1}", account.Name, newKeys.Secondary);
+                Console.WriteLine();
+
+                // Print subscription quota information
+                IPage<AccountResource> listResponse = await batchManagementClient.Account.ListAsync();
+                IList<AccountResource> accounts = listResponse.ToList();
+                Console.WriteLine("Total number of Batch accounts under subscription id {0}:  {1}", batchManagementClient.SubscriptionId, accounts.Count);
+
+                // Determine how many additional accounts can be created in the target region
+                int numAccountsInRegion = accounts.Count(o => o.Location == account.Location);
+                Console.WriteLine("Accounts in {0}: {1}", account.Location, numAccountsInRegion);
+                Console.WriteLine("You can create {0} more accounts in the {1} region.", quotaResponse.AccountQuota - numAccountsInRegion, account.Location);
+                Console.WriteLine();
+
+                // List accounts in the subscription
+                Console.WriteLine("Listing all Batch accounts under subscription id {0}...", batchManagementClient.SubscriptionId);
+                foreach (AccountResource acct in accounts)
+                {
+                    Console.WriteLine("  {0} - {1} | Location: {2}", accounts.IndexOf(acct) + 1, acct.Name, acct.Location);
+                }
+                Console.WriteLine();
+
+                // Delete account
+                Console.Write("Hit ENTER to delete account {0}: ", account.Name);
+                Console.ReadLine();
+                Console.WriteLine("Deleting account {0}...", account.Name);
+                try
+                {
+                    await batchManagementClient.Account.DeleteAsync(ResourceGroupName, account.Name);
+                }
+                catch (CloudException ex)
```

Comment:
Hmm, I feel like we need a comment here.  Something like this?

"Account deletion is a long running operation. This .DeleteAsync() method will submit the account deletion request and poll for the status of the long running operation until the account is deleted. Currently, querying for the operation status after the account is deleted will return a 404 error, so we have to add this catch statement. This behavior will be fixed in a future service release."

Alternatively, since this is a sample, we have a little more leeway in how we approach the problem than in our SDKs.  We could simplify this check to just continue on all 404s.  If the account never existed, it's arguable that calling the Delete method shouldn't be a terminating error.  We should probably still include the comment to explain where the 404 comes from though, since that's unexpected.

if (ex.Response.HttpStatusCode != HttpStatusCode.NotFound)
{
    throw;
}